src/controllers/hotels.controller.js

```diff
@@ -105,6 +105,54 @@ class Hotel {
 
     return Responses.handleSuccess(201, 'Room added successfully', res, room);
   }
+
+  /**
+   * Get all the hotels with likes
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} All hotels
+   */
+  async getAllHotels(req, res) {
+    const hotels = await hotelService.getHotels();
+    return Responses.handleSuccess(200, 'Hotels retrieved successfully', res, hotels);
+  }
+
+  /**
+   * Get hotel with likes and rooms
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} hotel
+   */
+  async getHotel(req, res) {
+    const { hotelId } = req.params;
+    const hotel = await hotelService.getHotelById(hotelId);
+    return Responses.handleSuccess(200, 'hotel retrieved successfully', res, hotel);
+  }
+
+  /**
+   * Like or unlike accomodation
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} response
+   */
+  async like(req, res) {
```

Comment:
This method is named `like` but I can see `unlike` operations also being performed. It can be renamed to something like `likeAndUnlike` but that violates the single responsibility principle. We might need a second handler to handle unliking.

Reply:
Fixed! thanks for feedback!